stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -232,6 +232,15 @@ public struct Unsafe${Mutable}BufferPointer<Element>
 %  end
   }
 
+  /// Creates an empty instance.
+  public init() {
+    // This is known to create a suitably-aligned pointer on all
+    // currently-supported platforms.
```

Comment:
Disallowing some values of Unsafe*Pointer would also make it impossible to represent (for example) some ObjC tagged pointer values.